app/services/ghost_to_strapi.rb

```diff
@@ -0,0 +1,335 @@
+        when "figure"
+          blocks << process_figure(child)
+        when "img"
+          blocks << process_image(child)
+        when "ul"
+          blocks << process_list(child, "unordered")
+        when "ol"
+          blocks << process_list(child, "ordered")
+        when "hr"
+          blocks << {
+            type: "paragraph",
+            children: [{
+              type: "text",
+              text: "---"
+            }]
+          }
+        when /h(\d)/
+          blocks << process_heading(child)
+        end
+      elsif child.comment?
+        if child.text == "kg-card-begin: html"
+          in_code_block = true
+          code_block_content = []
+        end
+      end
+    end
+    blocks.compact
+  end
+
+  def process_figure element
+    img = element.search("img").first
+    if img
+      src = img.attributes["src"].value
+      if src
+        process_image(src)
+      end
+    end
+  end
+
+  def process_image image_url
+    uri = URI.parse(image_url)
+    extension = File.extname(uri.path) || "png"
+    filename = "#{@current_slug}-#{@image_count}#{extension}"
+    strapi_image = check_strapi_for_image filename
+    unless strapi_image
+      begin
+        uri.open do |image|
+          File.binwrite("tmp/#{filename}", image.read)
+        end
+        response = JSON.parse(RestClient.post("#{@strapi_api}/upload",
+          {
+            files: File.open("tmp/#{filename}")
+          },
+          {
+            Authorization: "Bearer #{@strapi_api_key}"
+          }).body)
+        File.delete("tmp/#{filename}")
+        strapi_image = response.first
+      rescue OpenURI::HTTPError
+        strapi_image = nil
+      end
+    end
+
+    @image_count += 1
+    if strapi_image
+      {
+        type: "image",
+        image: strapi_image,
+        children: [{
+          type: "text",
+          text: ""
+        }]
+      }
+    end
+  end
+
+  def process_heading element
+    match = /h(\d)/.match(element.name)
+    {
+      type: "heading",
+      level: match[1],
+      children: [
+        {
+          text: element.content,
+          type: "text"
+        }
+      ]
+    }
+  end
+
+  def process_list element, type
+    {
+      type: "list",
+      format: type,
+      children: element.children.map {
+        {
+          type: "list-item",
+          children: [
+            {
+              type: "text",
+              text: _1.content
+            }
+          ]
+        }
+      }
+    }
+  end
+
+  def process_para content, type: "paragraph"
+    {
+      type:,
+      children: content.children.map { process_children(_1) }.compact
+    }
+  end
+
+  def process_children block
+    case block.name
+    when "a"
+      href = block.attributes["href"]&.value
+      if /^#.*/.match?(href)
+        {
+          type: "link",
+          url: "https://teachcomputing.org/blog/#{@current_slug}#{href}",
+          children: block.children.map { process_children(_1) }.compact
+        }
```

Comment:
Could you explain what is going on here?

Reply:
This has been removed now, fixed the problem in ghost rather than trying to man-handle it in conversion